File: CS/CustomTextEdit/CustomEdit.cs
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace CustomTextEdit
{
    public class MyCustomTextEdit : TextEdit
    {
        static MyCustomTextEdit()
        {
            RepositoryItemCustomTextEdit.Register();
        }

        public MyCustomTextEdit() { }

        public override string EditorTypeName
        {
            get { return RepositoryItemCustomTextEdit.CustomTextEditName; }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomTextEdit Properties
        {
            get { return base.Properties as RepositoryItemCustomTextEdit; }
        }
    }
}

File: VB/CustomTextEdit/CustomEdit.vb
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace CustomTextEdit

    Public Class MyCustomTextEdit
        Inherits TextEdit

        Shared Sub New()
            RepositoryItemCustomTextEdit.Register()
        End Sub

        Public Sub New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemCustomTextEdit.CustomTextEditName
            End Get
        End Property

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemCustomTextEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomTextEdit)
            End Get
        End Property
    End Class
End Namespace
